enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,269 @@
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: example
+      test: example-test
+    storage: 
+      spec:
+        storageClassName: hostpath
+        capacity:
+          storage: 1Gi
+        accessModes:
+          - ReadWriteOnce
+```
+
+#### Story 2
+A scorecard user can optionally specify storage class details within the scorecard configuration file to override storage defaults. Volume size, access mode, and storage class names can be specified as follows:
+```yaml
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: custom
+      test: example-test
+    storage: 
+      spec:
+        storageClassName: hostpath
+        capacity:
+          storage: 1Gi
+        accessModes:
+          - ReadWriteOnce
+```
+
+Valid values for access mode:
+ * ReadWriteOnce (default if not specified)
+ * ReadOnlyMany
+ * ReadWriteMany
+
+Access mode values are defined in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L556
+
+Valid values for volume size must be formatted according to the rules defined for storage 
+in https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Format.  Typical values for 
+storage are like:
+ * 1Gi (1 gigabyte) (default)
+ * 100M (100 megabyte)
+
+Storage classes can be found on a cluster by the following command:
+ * kubectl get sc
+
+A default storage class is the storage class on a cluster that has the following annotation:
+```yaml
+  metadata:
+    annotations:
+      storageclass.kubernetes.io/is-default-class: "true"
+```
+
+
+#### Story 3
+Scorecard gathers any persisted volume data upon test completion and stores it in a local directory where scorecard is executed.  Users
+can specify a non-default local path for test output using a command flag as follows:
+```bash
+operator-sdk scorecard ./bundle --selector=suite=custom --test-output=/mytestoutput
+```
+
+
+### Implementation Details/Notes/Constraints
+
+This feature, if specified by a scorecard user, requires a storage class be defined and operational on the k8s cluster nodes that scorecard tests would be executed upon.  If no storage class is specified or a default storage class is not available an error is returned for the scorecard test.
+
+### Risks and Mitigations
+
+Scorecard users that don't require persistent storage are not impacted by this proposed feature.
+
+## Design Details
+
+### Unique Volume Claims
+Each scorecard test that specifies the storage feature will have a unique PersistentVolumeClaim created for that test.  This insures that each
+test can be scheduled on any cluster node and that multiple tests will not co-mingle test output onto the same volume.  Scorecard deletes
+these volumes after test output has been gathered from the volume.  If the user specifies the command line flag of "--skip-cleanup", the volume will not
```

Comment:
`--skip-cleanup` also removes test pods. What if I just want storage persisted?

Reply:
this could be done if someone wants it for sure, it would mean a CLI flag be created for that purpose, something like --skip-storage-cleanup.

Reply:
Cleanup of everything or nothing is probably ok for a first pass. If there is demand for granular control then other flags can be added later. Just figured I'd ask in case I missed a reason why everything or nothing should be cleaned up.